### PR TITLE
MathProgNLSModel on mgh07 and mgh35

### LINKED ISSUE #14: MathProgNLSModel and user-defined functions
I implemented the mgh07 and mgh35 with MathProgNLSModel that use user-defined functions, nevertheless i have some issues.

For the first one, i find a way to implemented it with only one variable functions. But the MathProgNLSModel didn't find them. ```ERROR: KeyError: key :θ1 not found``` I think that ```MathProgNLSModel``` don't save/read the user-defined functions registered in the JuMP model.

For the second one, i need two variables user-defined functions. And actually JuMP and ForwardDiff package aren't able to calculate Hessian when we have more than one variable. The only workaround of that problem that i see is to give the function ```∇²f``` (if we can calculate it...) or calculate by FiniteDiff (i don't test it but I don't think that it's a good idea with non linear functions). If someone have a solution or an idea for that, it will be great.

The solution that i used in this case is to write the function ```Tsim(x,n)``` of mgh35 (in ```mgh35bis.jl``` file) under the form ```cos(n*Arccos(x))```, classic formula of Chebyshev polynomials.

[mgh07-35.zip](https://github.com/JuliaSmoothOptimizers/NLSProblems.jl/files/2041414/mgh07-35.zip)

### PR DESCRIPTION
**Pull request by Xx-fractale-xX**
resolved #14

### CODE REVIEW

File: REQUIRE [hunk 1]
```diff
@@ -1,2 +1,2 @@
 julia 0.6
-NLPModels 0.4.1
+NLPModels
```
Comment (dpo) [inline REQUIRE]: It's recommended to specify a version here. Maybe we need a new release?

Comment (Xx-fractale-xX) [reply REQUIRE]: Yes, I deleted the version to see if the tests passed. But the continuous integration install the last tagged version of ```NLPModels``` even so.

**Comment by dpo** [comment]
@abelsiqueira Do we need a new release of NLPModels here?

**Comment by abelsiqueira** [comment]
v0.4.2 release for user defined functions. https://github.com/JuliaLang/METADATA.jl/pull/15216

**Comment by abelsiqueira** [comment]
v0.4.2 merged.

**Comment by dpo** [comment]
Thank you both!

**Pull request merged**